presto-main/src/main/java/com/facebook/presto/execution/QueryTracker.java

```diff
@@ -409,6 +414,20 @@ public interface TrackedQuery
         void fail(Throwable cause);
 
         // XXX: This should be removed when the client protocol is improved, so that we don't need to hold onto so much query history
-        void pruneInfo();
+        void pruneInfo(PruneEvent event);
+    }
+
+    public enum PruneEvent
```

Comment:
I would just add two methods, rather than introduce a prune event concept.  Perhaps `pruneExpiredQueryInfo` and `pruneFinishedQueryInfo`?

Reply:
updated